### PR TITLE
VCD annotation fixes

### PR DESCRIPTION
**Pull request by alaindargelas**

### CODE REVIEW

**Comment by alaindargelas** [comment]
@akashlevy , please review

File: power/Power.cc [hunk 1]
```diff
@@ -1204,7 +1204,14 @@ Power::findActivity(const Pin *pin)
...
     if (activity.origin() != PwrActivityOrigin::unknown)
       return activity;
+  } else if (user_activity_map_.hasKey(pin)) {
+    PwrActivity &activity = user_activity_map_[pin];
+    if (activity.origin() != PwrActivityOrigin::unknown)
+      return activity;
   }
+  const char *pname = network_->pathName(pin);
+  if (strstr(pname, "/IQ") == nullptr)
+    std::cout << "WARNING: No Activity found for: " << pname << "\n";
```
Comment (akashlevy) [inline power/Power.cc:1214]: Can you use the native OpenSTA warning system instead? Cherry wouldn't accept this upstream.

File: power/Power.cc [hunk 2]
```diff
@@ -1204,7 +1204,14 @@ Power::findActivity(const Pin *pin)
     PwrActivity &activity = activity_map_[pin];
     if (activity.origin() != PwrActivityOrigin::unknown)
       return activity;
+  } else if (user_activity_map_.hasKey(pin)) {
+    PwrActivity &activity = user_activity_map_[pin];
+    if (activity.origin() != PwrActivityOrigin::unknown)
+      return activity;
   }
+  const char *pname = network_->pathName(pin);
+  if (strstr(pname, "/IQ") == nullptr)
```
Comment (akashlevy) [inline power/Power.cc:1213]: Could we instead check whether the pin is a "real" pin or an internal pin? That would be a more general solution than this.

Comment (akashlevy) [reply power/Power.cc:1213]: At the very least put a TODO

File: power/Power.cc [hunk 3]
```diff
@@ -1204,7 +1204,14 @@ Power::findActivity(const Pin *pin)
     PwrActivity &activity = activity_map_[pin];
     if (activity.origin() != PwrActivityOrigin::unknown)
       return activity;
+  } else if (user_activity_map_.hasKey(pin)) {
```
Comment (akashlevy) [inline power/Power.cc:1207]: This annotates the net activity on to the pin, right? Just for my understanding.

File: power/SaifReader.cc [hunk 4]
```diff
@@ -191,6 +192,7 @@ SaifReader::setNetDurations(const char *net_name,
                    tc,
                    activity);
         power_->setUserActivity(pin, activity, duty, PwrActivityOrigin::saif);
+        std::cout << "SAIF Annotated pin: " << net_name << " tr. count: " << tc <<  " activity: " << activity << " duty: " <<  duty << std::endl;
```
Comment (akashlevy) [inline power/SaifReader.cc:195]: This should use a debug logger instead. Are we even supporting SAIF at this point anyways?

File: power/ReadVcdActivities.cc [hunk 5]
```diff
@@ -175,13 +176,41 @@ ReadVcdActivities::setVarActivity(VcdVar *var,
...
+
 void
 ReadVcdActivities::setVarActivity(const char *pin_name,
                                   const VcdValues &var_values,
                                   int value_bit)
 {
-  const Pin *pin = sdc_network_->findPin(pin_name);
-  if (pin) {
+  std::vector<const Pin*> pins;
+  std::string pinname = replaceAll(pin_name, "\\", "");
```
Comment (akashlevy) [inline power/ReadVcdActivities.cc:196]: Cherry has a function that does something similar to this already in the string utilities I believe

File: power/ReadVcdActivities.cc [hunk 6]
```diff
@@ -196,6 +225,13 @@ ReadVcdActivities::setVarActivity(const char *pin_name,
       checkClkPeriod(pin, transition_count);
     power_->setUserActivity(pin, activity, duty, PwrActivityOrigin::vcd);
     annotated_pins_.insert(pin);
+    //const char *pname = cmd_network_->pathName(pin);
+    //std::cout << "VCD Annotated pin: " << pin_name  << " port: " << pname << " tr count: " << transition_count
+    //          << " activity: " << activity << " duty: " << duty << std::endl;
+  }
+
+  if (pins.empty()) {
+    std::cout << "WARNING: Missed annotating pin: " << pin_name << std::endl;
```
Comment (akashlevy) [inline power/ReadVcdActivities.cc:234]: Use native warning

File: power/Power.cc (hunk 3 above)
Comment (alaindargelas) [reply power/Power.cc:1207]: The activity_map_ used originally does not contain all the pins, on the other hand, the user_activity_map_ does contain all the pins and is needed to be able to calculate the Power on all the cells. I found that through gdb debugging. Maybe the activity_map_ was meant to be  user_activity_map_, something to check with Cherry.

File: power/Power.cc (hunk 2 above)
Comment (alaindargelas) [reply power/Power.cc:1213]: Adding a TODO to remove this code altogether, this is internal debugging while in dev mode.

File: power/Power.cc (hunk 1 above)
Comment (alaindargelas) [reply power/Power.cc:1214]: Adding a TODO to remove this code altogether, this is internal debugging while in dev mode.

File: power/ReadVcdActivities.cc (hunk 6 above)
Comment (alaindargelas) [reply power/ReadVcdActivities.cc:234]: Adding a TODO to remove this code altogether, this is internal debugging while in dev mode.

File: power/ReadVcdActivities.cc (hunk 5 above)
Comment (alaindargelas) [reply power/ReadVcdActivities.cc:196]: None that I can find

**Comment by alaindargelas** [comment]
This PR is not needed anymore

**Pull request closed without merging**